### PR TITLE
Add ArrayBufferFIleReader supperted

### PR DESCRIPTION
**Pull request by TongDaDa**
Now, for my updated codes, It just supports the type of ArrayBuffer, but It also can support Uint8Array and other Class about binary by get `buffer` property of them. What do you think?

### CODE REVIEW

**Comment by TongDaDa** [comment]
Your codes are very beautiful, I learned a lot of design patterns of the programme.

**Comment by aadsm** [comment]
@LiuTongTong in that code we're trying to figure out which media tag reader is able to read the metadata from the audio file.
To do that each media tag reader reads a specific section of the file looking for a signature that will identify the metadata.
We don't want to read the entire audio file just for that so we ask each tag reader what are the byte ranges where the signature exists, that's what `getTagIdentifierByteRange` returns. We then only load the part of the file that has those byte ranges.
The if/continue you mention acts as a check to make sure the audio file itself is big enough for us to read the byte rages where the signature should be. For example, imagine the signature is in bytes [128-130], if the entire file is only 100bytes then we can't read that range so we skip that, that's why we continue.

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aadsm) [inline .idea/vcs.xml]: Can you remove this file from the pull request? It's specific to the editor you used.

File: src/ArrayBufferFileReader.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        this._fileData = new ChunkedFileData();
+    }
+
+    static canReadFile(file: any): boolean {
+        return typeof ArrayBuffer === 'function' && file instanceof ArrayBuffer
+    }
+
+    _init(callbacks: LoadCallbackType): void {
+        this._size = this._buffer.byteLength;
+        console.log(this._size);
```
Comment (aadsm) [inline src/ArrayBufferFileReader.js]: don't forget to remote this console.log.

Comment (TongDaDa) [reply src/ArrayBufferFileReader.js]: Ok

**Comment by TongDaDa** [comment]
Tanks for your suggest!

**Comment by TongDaDa** [comment]
How  do you think? i  had changed these files

**Comment by aadsm** [comment]
@TongDaDa hey, last weekend I ended up working on another project I have so didn’t have time to review this, sorry about that! I’ll check it this weekend, thank you for contributing!

**Comment by aadsm** [comment]
I’m able to understand your English just fine, it’s not a problem. I’m not a native either :-).

**Comment by TongDaDa** [comment]
Ok, I like to contribute some open-source projects, so could I develop opened-projects with you?

**Comment by aadsm** [comment]
> Ok, I like to contribute some open-source projects, so could I develop opened-projects with you?

Yes, that would be great!

**Comment by TongDaDa** [comment]
> > Ok, I like to contribute some open-source projects, so could I develop opened-projects with you?
> 
> Yes, that would be great!

hahahaa, that is great, I'll learn a lot of knowledge.

**Comment by aadsm** [comment]
I've just merged and also added you to the list of contributors! https://github.com/aadsm/jsmediatags/blob/master/CONTRIBUTORS.md

**Pull request merged**